src/main/java/graphql/analysis/QueryTraversal.java

```diff
@@ -256,7 +310,9 @@ public TraversalControl visitFragmentSpread(FragmentSpread fragmentSpread, Trave
             QueryTraversalContext parentEnv = context.getVarFromParents(QueryTraversalContext.class);
 
             GraphQLCompositeType typeCondition = (GraphQLCompositeType) schema.getType(fragmentDefinition.getTypeCondition().getName());
-
+            Assert.assertTrue(typeCondition != null, "Invalid type condition '%s' in fragment '%s'", fragmentDefinition.getTypeCondition().getName(),
```

Comment:
we have `Assert.assertNonNull`

Reply:
but this assert is not really anyway, because we assume the query is a valid one, right?

Reply:
I am not sure. If you put wrong query (e.g. with fragment that is not suppose to be there) you will get weird NPE that will not tell you what's going on.